grammar.js

```diff
@@ -954,9 +957,38 @@ module.exports = grammar({
       field('type_arguments', $.type_arguments)
     )),
 
-    call_expression: $ => prec(PREC.call, seq(
-      field('function', $._simple_expression),
-      field('arguments', choice($.arguments, $.case_block, $.block)),
+    call_expression: $ => choice(
+      prec.left(PREC.call, seq(
+        field('function', $._simple_expression),
+        field('arguments', choice(
+          $.arguments,
+          $.case_block,
+          $.block,
+        )),
+      )),
+      prec.right(PREC.colon_call, seq(
+        field('function', $._postfix_expression_choice),
+        ':',
+        field('arguments', $.colon_argument),
+      )),
+    ),
+
+    colon_argument: $ => prec.left(PREC.colon_call, seq(
+      optional(field('lambda_start', seq(
+        choice(
+          $.bindings,
+          $._identifier,
+          $.wildcard,
+        ),
+        '=>',
+      ))),
+      $._colon_argument_block,
+    )),
+
+    _colon_argument_block: $ => prec.left(seq(
+      $._indent,
+      field('body', $._block),
```

Comment:
Should we have `case_block` alternative here?

Seems to be allowed:

```scala
scala> import language.experimental.fewerBraces

scala> List(1,2,3,4,5).collect:
     |   case i => i + 1
     |
val res0: List[Int] = List(2, 3, 4, 5, 6)
```

Reply:
Nice catch!
```
  ColonArgument     ::=  colon [LambdaStart]
                         (CaseClauses | Block)
```

Reply:
Done - https://github.com/tree-sitter/tree-sitter-scala/pull/128/commits/a7e939493522d04b4efad2ca57d0977c1513d330